bokeh/models/ranges.py

```diff
@@ -90,6 +100,19 @@ class DataRange1d(DataRange):
     automatically computed end value.
     """)
 
+    bound_lower = Float(help="""
+    The minimum value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data. If not provided, will
+    default to the min of the data.
+    """)
```

Comment:
I just realized I might have misunderstood your comment.

I got the type from `min`. 

Are you saying that the accepted values for `min`, `max`, `bound_lower`, and `bound_upper` on `DataRange1d` should be `Either(Auto, Float)`.